src/nrniv/CMakeLists.txt

```diff
@@ -9,42 +9,71 @@ add_compile_options(${NRN_COMPILE_WARNING_SUPPRESSIONS})
 # =============================================================================
 # Build modlunit : Mod file units checker
 # =============================================================================
-flex_target(modlunitlexer ${NRN_MODLUNIT_SRC_DIR}/lex.lpp ${NRN_MODLUNIT_SRC_DIR}/lex.cpp)
-
-bison_target(modlunitparser ${NRN_MODLUNIT_SRC_DIR}/parse1.ypp ${NRN_MODLUNIT_SRC_DIR}/parse1.cpp
-             DEFINES_FILE ${NRN_MODLUNIT_SRC_DIR}/parse1.hpp)
-
-set_property(
-  SOURCE ${NRN_MODLUNIT_SRC_FILES}
-  APPEND
-  PROPERTY COMPILE_DEFINITIONS NRNUNIT=1)
-
-add_executable(modlunit ${NRN_MODLUNIT_SRC_FILES})
-
-if (NRN_NMODL_CXX_FLAGS)
+set(NRN_MODLUNIT_GEN "${CMAKE_CURRENT_BINARY_DIR}/modlunit_generated")
+file(MAKE_DIRECTORY "${NRN_MODLUNIT_GEN}")
+file(RELATIVE_PATH NRN_MODLUNIT_SRC_REL "${NRN_MODLUNIT_GEN}" "${NRN_MODLUNIT_SRC_DIR}")
+# Run flex and bison with relative paths, so absolute paths are not present in the generated source
+# file because of include, __FILE__ and so on. This improves ccache performance.
+add_custom_command(
+  OUTPUT "${NRN_MODLUNIT_GEN}/lex.cpp"
+  WORKING_DIRECTORY "${NRN_MODLUNIT_GEN}"
+  COMMAND "${FLEX_EXECUTABLE}" ARGS -o lex.cpp "${NRN_MODLUNIT_SRC_REL}/lex.lpp"
+  DEPENDS "${NRN_MODLUNIT_SRC_DIR}/lex.lpp"
+  COMMENT "[FLEX][modlunitlexer] Building scanner with flex ${FLEX_VERSION}")
+add_custom_command(
+  OUTPUT "${NRN_MODLUNIT_GEN}/parse1.hpp" "${NRN_MODLUNIT_GEN}/parse1.cpp"
+  WORKING_DIRECTORY "${NRN_MODLUNIT_GEN}"
+  COMMAND "${BISON_EXECUTABLE}" ARGS --defines=parse1.hpp -o parse1.cpp
+          "${NRN_MODLUNIT_SRC_REL}/parse1.ypp"
+  DEPENDS "${NRN_MODLUNIT_SRC_DIR}/parse1.ypp"
+  COMMENT "[BISON][modlunitparser] Building parser with bison ${BISON_VERSION}")
+
+add_executable(modlunit ${NRN_MODLUNIT_SRC_FILES} "${NRN_MODLUNIT_GEN}/lex.cpp"
+                        "${NRN_MODLUNIT_GEN}/parse1.cpp")
+target_compile_definitions(modlunit PRIVATE NRNUNIT=1)
+# Generated .cpp needs to find source-directory .hpp and vice versa.
+target_include_directories(modlunit PRIVATE "${NRN_MODLUNIT_SRC_DIR}" "${NRN_MODLUNIT_GEN}")
+
```

Comment:
@olupton  : A quick comment: as this is repeated pattern, is there way to hide this behind cmake macro / function? 🤔

Reply:
I'm sure it's possible. With the same pattern appearing in NEURON, CoreNEURON, NMODL and MOD2C I guess the natural home would be `hpc-coding-conventions` (and leaving it explicit in MOD2C if the coding conventions are not included there). Do you think that sounds worthwhile?

Reply:
Don't have strong opinion: it's not blocker for this OR. Going this in hpc-coding-convention means even standalone/vanilla installation of neuron would require submodule clone (i.e. zip from GitHub would be non-functional for basic installation). Hence I am of two minds.

Reply:
I would lean towards leaving it as it is. But if others have strong opinions, I can change it.

Reply:
Sounds ok to me.